kedro-datasets/kedro_datasets/spark/spark_stream_dataset.py

```diff
@@ -0,0 +1,158 @@
+"""SparkStreamingDataSet to load and save a PySpark Streaming DataFrame."""
+import os
+from typing import Any, Dict
+from copy import deepcopy
+import yaml
+from kedro.io import AbstractDataSet
+from pyspark import SparkConf
+from pathlib import PurePosixPath
+from pyspark.sql import SparkSession, DataFrame
+from yaml.loader import SafeLoader
+from kedro_datasets.spark.spark_dataset import _split_filepath
+
+class SparkStreamingDataSet(AbstractDataSet):
+    """``SparkStreamingDataSet`` loads data into Spark Streaming Dataframe objects.
+
+    Example usage for the
+    `YAML API <https://kedro.readthedocs.io/en/stable/data/\
+    data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
+    .. code-block:: yaml
+
+        raw.new_inventory:
+            type: streaming.extras.datasets.spark_streaming_dataset.SparkStreamingDataSet
```

Comment:
```suggestion
            type: spark.SparkStreamingDataSet
```

`SparkStreamingDataSet should be added into the `spark/__init__.py` and imported from there.